filesystems/demo-dos/src/main/scala/cromwell/filesystems/demo/dos/DemoDosPathBuilder.scala

```diff
@@ -0,0 +1,73 @@
+package cromwell.filesystems.demo.dos
+
+import com.google.api.client.testing.http.MockHttpTransport
+import com.google.api.client.testing.json.MockJsonFactory
+import com.google.api.services.storage.Storage
+import com.google.cloud.storage.StorageOptions
+import com.google.cloud.storage.contrib.nio.{CloudStorageConfiguration, CloudStoragePath}
+import cromwell.core.path.{NioPath, Path, PathBuilder}
+import cromwell.filesystems.demo.dos.DemoDosPathBuilder._
+import cromwell.filesystems.gcs.{GcsPath, GcsPathBuilder}
+
+import scala.util.{Failure, Success, Try}
+
+class DemoDosPathBuilder extends PathBuilder {
+  private val mockStorage: Storage =
+    new Storage.Builder(new MockHttpTransport, new MockJsonFactory, null)
```

Comment:
OOC why a mock storage and not a real one here ?
Is it going to work with all types of Credentials passed from workflow options etc.. ?

Reply:
The Demo DOS file system implementation cannot and does not actually talk to the cloud for accessing file contents/listing. Plus, it only supports a subset of path manipulation. Both path-without-scheme & path-as-string are supported. Unsupported operations include resolve-sibling & get-parent.

Back to accessing DOS file data, the mocks should never be invoked by Cromwell. This demo implementation wraps GCS as it is the most stable file system, but could have wrapped S3 or OSS. Or I could have copy/pasted yet another filesystem beyond the three copies we have already. Ideally woodard will halt the need to blindly copy `java.nio.Path` or `cromwell.core.path.Path` implementations in the future. 😉